### PR TITLE
Create pixel activation heatmap

### LINKED ISSUE #7: Create pixel activation heatmap
Depends on: #4, #6 

*5.4.3 Visualizing heatmaps of class activation* of [Deep learning with Python](https://tanthiamhuat.files.wordpress.com/2018/03/deeplearningwithpython.pdf).

Most recent version of this example from [Keras website](https://keras.io/examples/vision/grad_cam/).

### PR DESCRIPTION
**Pull request by SzymonZos**
Closes #7

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -138,3 +138,9 @@ dmypy.json
 client_secrets.json
 credentials.json
 settings.yaml
+
+# Logs
+log/
```
Comment (maciejzj) [inline .gitignore:143]: I didn't include this `log` for purpose. We should store crucial models and logs from their training somewhere. For example this model you crated and trained in this MR (with accuracy of 77%) should be saved somewhere. There are pro ways to store models and data about them, which provide tools for models comparison and management (https://mlflow.org), however using them costs effort (and maybe a server). Simply adding models directly to git is ok for the time being, because it is effortless.

File: make_models.py [hunk 2]
```diff
@@ -3,11 +3,12 @@
...
 from tensorflow.keras import layers
 from tensorflow.keras import models
+from tensorflow.keras import regularizers
 
 
 def save_model(model: models.Model, name: str, timestamp=True):
-    timestamp = datetime.now().strftime("%y-%m-%d-%H:%M:%S")
-    model.save('log/models/model_' + timestamp + ' ' + name)
+    timestamp = datetime.now().strftime("%y-%m-%d_%H_%M_%S")
+    model.save('log/models/model_' + timestamp + '_' + name)
```
Comment (maciejzj) [inline make_models.py:11]: Don't see a reason to change time format to underscores, but if you will it can stay. Changing space to underscore is big plus. Dunno how the space managed to slip in here.

File: make_models.py [hunk 3]
```diff
@@ -17,17 +18,33 @@ def load_model(name: str) -> models.Model:
...
 def make_simple_cnn(input_shape=(640, 640, 3)) -> models.Model:
     model = models.Sequential()
+    l2_regularizer = regularizers.l2(0.001)
 
-    model.add(layers.Conv2D(6, (3, 3), activation='relu',
-                            input_shape=input_shape))
+    model.add(layers.Conv2D(32, (3, 3), kernel_regularizer=l2_regularizer,
+                            activation='relu', input_shape=input_shape))
     model.add(layers.MaxPool2D())
+    # model.add(layers.Dropout(0.2))
```
Comment (maciejzj) [inline make_models.py]: This many dropouts were overkill idea. You can delete the ones that are commented out.

File: .gitignore (hunk 1 above)
Comment (SzymonZos) [reply .gitignore:143]: This directory is full of mess. Do we really need all the files?

File: make_models.py (hunk 3 above)
Comment (SzymonZos) [reply make_models.py]: Sure, forgot about that.

File: make_models.py (hunk 2 above)
Comment (SzymonZos) [reply make_models.py:11]: The reason is pretty simple - it doesn't work with `:`

File: .gitignore (hunk 1 above)
Comment (maciejzj) [reply .gitignore:143]: This may seem mess, but these files are tensorboard data. Don’t tinker with them directly, tensorboard creates nice visualizations from them.

Ofc you should push only models and logs that are worth saving.

File: main.py [hunk 4]
```diff
@@ -36,7 +38,14 @@
     }
 
     model = make_simple_cnn()
-    img_path = "../images/train/true/IMG_6605.jpg"
     train(model, params)
-    save_model(model, "dupa")
-    plot_heatmaps(img_path, params['img_size'], model, "dupa.jpg")
+    save_model(model, "first_working_model")
```
Comment (maciejzj) [inline main.py]: We have callback for saving best model, so explicit save isn't necessary (what's more it may be worse, because it saves last state of model, not best one).

If there is a reason for saving model directly use `args.name` for its name.

File: .gitignore (hunk 1 above)
Comment (SzymonZos) [reply .gitignore:143]: Since logs contain some very big data, I will leave it in gitignore for now

File: main.py (hunk 4 above)
Comment (SzymonZos) [reply main.py]: Done
